### PR TITLE
Round up elem_arr_1 factor for index tensors

### LINKED ISSUE #4047: SDSC elem_arr_1 factor floors to 0 for sub-stick dimensions
### What happened?

We're in the processing of formalizing the SDSC spec and ran into this.

`codegen/compute_ops.py:414-417` uses floor division with no lower bound:

    "factor_": 1 if is_stick_reduction else (size // elems_per_stick),

When `size < elems_per_stick` this yields 0. Repro:

```python
import torch
import torch_spyre  # noqa: F401

M, N, P = 8, 4, 3

out = torch.rand(M, N, dtype=torch.float16).to("spyre")
src = torch.rand(P, N, dtype=torch.float16).to("spyre")
idx = torch.arange(P, dtype=torch.int32).to("spyre")

@torch.compile
def kernel(out, src, idx):
    return out.index_add_(0, idx, src)

kernel(out, src, idx)
```

The int32 index tensor is 1-D with mb as its own stick dim (stickSize_: [32]) and extent P=3, so 3 // 32 == 0:

```
{"factor_": 0,  "label_": "elem_arr_1"},
{"factor_": 32, "label_": "elem_arr_0"}
```

Also reproduces via
tests/inductor/test_indirect_access_scatter.py::TestScatter_cores32::test_index_add_moe_shape
(M=64, N=2816, P=3), and a few other test cases.

My understanding is factor_ indicates how many valid coordinate values for the corresponding dim_prop_func, so essentially 0 <= x <= (factor_ - 1) which 0 shouldn't be valid.  I believe the number should be rounded up instead of rounded down.

### What did you expect to happen?

I discussed this with Alberto, who agrees a factor_ of 0 does not make sense here. Three elements should occupy one partially-filled stick, so the expected value is 1. The sibling is_stick_reduction branch on the same line already returns 1 rather than 0 for its own degenerate case.

### How can we reproduce it?

```
import torch
import torch_spyre  # noqa: F401

M, N, P = 8, 4, 3

out = torch.rand(M, N, dtype=torch.float16).to("spyre")
src = torch.rand(P, N, dtype=torch.float16).to("spyre")
idx = torch.arange(P, dtype=torch.int32).to("spyre")

@torch.compile
def kernel(out, src, idx):
    return out.index_add_(0, idx, src)

kernel(out, src, idx)
```

and observe sdsc_4.json

### Any environmental details we need to know?

none

### Anything else we need to know?

_No response_

### Relevant log output

```Shell

```

### PR DESCRIPTION
**Pull request by avery-blanchard**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [x] bug
- [ ] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

Fix floor-division bug that produces invalid factor for `elem_arr_1` when tensor stick dimension are smaller than `elems_per_stick` (index tensors).

#### Which issue(s) this PR is related to:
Fixes #4047

### CODE REVIEW

File: torch_spyre/_inductor/codegen/compute_ops.py [hunk 1]
```diff
@@ -413,7 +413,11 @@ def gen_coord_info_value(
                     {
                         "factor_": 1
                         if is_stick_reduction
-                        else (size // elems_per_stick),
+                        else (
+                            size // elems_per_stick
+                            if size % elems_per_stick == 0
```
Comment (tinhto-000) [inline torch_spyre/_inductor/codegen/compute_ops.py]: @avery-blanchard this might be faster and more pythonic:

```
@@ -413,7 +413,7 @@ def gen_coord_info_value(
                     {
                         "factor_": 1
                         if is_stick_reduction
-                        else (size // elems_per_stick),
+                        else -(-size // elems_per_stick),
                         "label_": "elem_arr_1",
                     },
```

File: torch_spyre/_inductor/codegen/compute_ops.py
Comment (tinhto-000) [inline torch_spyre/_inductor/codegen/compute_ops.py]: @avery-blanchard do you think we might have the same issue here?

Comment (avery-blanchard) [reply torch_spyre/_inductor/codegen/compute_ops.py]: I added it for safety, but we shouldn't hit a problem here

**Review by dgrove-oss — APPROVED** [review]
lgtm
